stdlib/random.ml

```diff
@@ -93,6 +93,39 @@ module State = struct
     then invalid_arg "Random.int"
     else intaux s bound
 
+  let rec int63aux s n =
+    let max_int_32 = (1 lsl 30) + 0x3FFFFFFF in (* 0x7FFFFFFF *)
+    let b1 = bits s in
+    let b2 = bits s in
+    let (b, max_int) =
+      if n <= max_int_32 then
+        (* 32 random bits on both 64-bit OCaml and JavaScript.
+           Use upper 16 bits of b1 and b2. *)
+        let bpos =
+          (((b2 land 0x3FFFC000) lsl 2) lor (b1 lsr 14)) land max_int_32
```

Comment:
I missed the `land max_int_32` in my previous reviews but I'm afraid it's wrong.  The effect is to zero bit number 31, but the `let r = b lsr 1` below will then zero bit number 30, and you end up with a candidate between 0 and 2^30 instead of between 0 and 2^31, and random values above 2^30 will never be generated.

Reply:
I took the liberty of pushing to your branch a new test that performs chi-square testing on the functions from module Random.  You'll see that the test `"Random.int63 (256 * p) / p"` fails the test.  It exercises the case 2^30 < n < 2^31.

Reply:
I updated the test following @Octachron 's comments (concerning the confidence interval) and to test the js-of-ocaml model (integer size = 32 bits, word size = 32 bits) as well.